llvm/lib/DebugInfo/DWARF/DWARFDebugLine.cpp

```diff
@@ -1389,16 +1397,36 @@ bool DWARFDebugLine::LineTable::lookupAddressRangeImpl(
   Sequence.SectionIndex = Address.SectionIndex;
   Sequence.HighPC = Address.Address;
   SequenceIter LastSeq = Sequences.end();
-  SequenceIter SeqPos = llvm::upper_bound(
-      Sequences, Sequence, DWARFDebugLine::Sequence::orderByHighPC);
-  if (SeqPos == LastSeq || !SeqPos->containsPC(Address))
-    return false;
+  SequenceIter SeqPos;
+
+  if (StmtSequenceOffset) {
+    // If we have a statement sequence offset, find the specific sequence
+    // Binary search for sequence with matching StmtSeqOffset
+    Sequence.StmtSeqOffset = *StmtSequenceOffset;
+    SeqPos = std::lower_bound(Sequences.begin(), LastSeq, Sequence,
```

Comment:
My bad here - Sequences is sorted by high pc, instead of by offset. So this'll have to be a linear search - so just replacing this lower_bound with llvm::find_if.

And rather than using the `Sequence` object, using llvm::find_if you can search for the thing with the desired StmtSeqOffset directly comparing it to *StmtSequenceOffset.
```
SeqPos = llvm::find_if(Sequences, [&](const DWARFDebugLine::Sequence &Seq) {
  return Seq.StmtSeqOffset == *StmtSequenceOffset;
});
...
```